### PR TITLE
Add Action Summary (home) and Alert (other pages)

### PR DESCRIPTION
**Pull request by john12321**
Work so far:

- Add an Action Summary section to TSS home page
- Add a Action Summary Alert to be displayed on all pages except Home and MFA (when `NOMFA` ) with text indicating outstanding action (CoJ) or other important Form R submission-related info.
-  Minimise TSS Header on mobile so you can at least see the Action Summary header  

Note: The Action Summary Alert state management (mainly around the Form R lists) is a bit messy so will most probably create a follow-up ticket to refactor this.

TIS21-5400
TIS21-5508

TODO 
Once this is merged, Tosin will add the Hotjar surveys to the CoJ and Form R confirm pages https://hee-tis.atlassian.net/browse/TIS21-5514

### CODE REVIEW

File: components/home/actionSummary/ActionSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                        <Label size="s">
+                          <FontAwesomeIcon
+                            icon={faExclamationCircle}
+                            color="#DA291C"
+                            size="lg"
+                          />{" "}
+                          You have {unsignedCojCount} unsigned{" "}
+                          <Link to="/programmes">
+                            {`Conditions of Joining Agreement${
+                              unsignedCojCount > 0 ? "s" : ""
```
Comment (ReubenRobertsHEE) [inline components/home/actionSummary/ActionSummary.tsx:74]: >1 I think ;)

**Comment by ReubenRobertsHEE** [comment]
Perhaps not a priority edge-case to deal with, but if the forms service is down, the messaging is a little misleading:
image

File: components/main/GlobalAlert.tsx [hunk 2]
```diff
@@ -69,24 +91,61 @@ function BookmarkAlert() {
...
+  inProgressFormR,
+  importantInfo
+}: Readonly<OutstandingActionsAlertProps>) {
+  const ACTION_LINK = (
+    <span>
+      Please click <Link to="/">here</Link> for details.
+    </span>
+  );
+  const importantInfoText =
+    "You have important information about your Form R submission dates you might want to check.";
```
Comment (ReubenRobertsHEE) [inline components/main/GlobalAlert.tsx:112]: Maybe something a bit less wordy (and polite, ugh :>), like 'Please review your Form R submissions'?

File: components/main/GlobalAlert.tsx [hunk 3]
```diff
@@ -6,6 +6,9 @@ import { useInfoActions } from "../../utilities/hooks/action-summary/useInfoActi
 import { useInProgressActions } from "../../utilities/hooks/action-summary/useInProgressActions";
 
 const GlobalAlert = () => {
+  // Don't show the Global alert if the user has not set their MFA
```
Comment (ReubenRobertsHEE) [inline components/main/GlobalAlert.tsx:9]: Why not man?

Comment (ReubenRobertsHEE) [reply components/main/GlobalAlert.tsx:9]: (Ok, ignore, dealt with in later commits I think)

**Comment by ReubenRobertsHEE** [comment]
The SonarCloud stuff seems irrelevant (applies to existing code). I haven't checked the code coverage on the tests.

**Review by ReubenRobertsHEE — DISMISSED** [review]
Nothing in the PR comments that can't be picked up in the future, if they apply at all. 🤞

**Comment by john12321** [comment]
> Perhaps not a priority edge-case to deal with, but if the forms service is down, the messaging is a little misleading: image

Yup, not great UX . As discussed, will tackle this in a follow-up ticket (which includes other similar instances e.g. a forms service failure on forms list page).

File: utilities/ActionSummaryUtilities.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+export function isLatestSubmissionDateWithinLastYear(
+  lastSubStartDate: DateType
+): boolean {
+  const lastSubDate = dayjs(lastSubStartDate);
+  const today = dayjs();
+  const oneYearAgo = today.subtract(1, "year");
+  return (
+    lastSubDate.isAfter(oneYearAgo) &&
+    (lastSubDate.isBefore(today) || lastSubDate.isSame(today))
```
[the remark this answers is not in the document]
Comment (john12321) [reply utilities/ActionSummaryUtilities.ts]: As discussed, will tackle in follow-up ticket - after we've (well you mostly! 😁) have established a failure case (taking into account possible BE standardisation to server time).

File: utilities/__test__/ActionSummary.test.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    const lastSubStartDate = dayjs().subtract(1, "year").toDate();
+    expect(isLatestSubmissionDateWithinLastYear(lastSubStartDate)).toBe(false);
+  });
+  it("should return false when latest submission date is more than a year ago.", () => {
+    const lastSubStartDate = dayjs()
+      .subtract(1, "year")
+      .subtract(1, "day")
+      .toDate();
+    expect(isLatestSubmissionDateWithinLastYear(lastSubStartDate)).toBe(false);
+  });
```
[the remark this answers is not in the document]
Comment (john12321) [reply utilities/__test__/ActionSummary.test.ts]: (see above comment)

**Comment by john12321** [comment]
> Nothing in the PR comments that can't be picked up in the future, if they apply at all. 🤞

Will just sort the working-related 'quick fixes' now.

**Comment by john12321** [comment]
so much for the quickness of the fixes!

**Review by ReubenRobertsHEE — APPROVED** [review]

**Pull request merged**